math/src/elliptic_curve/short_weierstrass/point/jacobian.rs

```diff
@@ -0,0 +1,132 @@
+//! Jacobian variant of projective
+//! coordinates. It works in the similar way, but differs in how affine get divided by $z$ ($\frac{x}{z^2}, \frac{y}{z^3}$).
+
+use crate::{
+    cyclic_group::IsGroupElement, elliptic_curve::short_weierstrass::traits::IsShortWeierstrass,
+    field::element::FieldElement,
+};
+
+use super::ShortWeierstrassProjectivePoint;
+
+// feels like `crate::elliptic_curbe::point` could be redesigned to provide the trait to implement here
+#[derive(Clone)]
+pub struct JacobianPoint<E: IsShortWeierstrass> {
+    x: FieldElement<E::BaseField>,
+    y: FieldElement<E::BaseField>,
+    z: FieldElement<E::BaseField>,
+}
+
+impl<E: IsShortWeierstrass> PartialEq for JacobianPoint<E> {
+    fn eq(&self, other: &Self) -> bool {
+        let (z_sq_self, z_sq_other) = (self.z.square(), other.z.square());
+        (&self.x * &z_sq_self == &other.x * &z_sq_other)
+            && (&self.y * &z_sq_self * &self.z == &other.y * z_sq_other * &other.z)
+    }
+}
+impl<E: IsShortWeierstrass> Eq for JacobianPoint<E> {}
+
+impl<E: IsShortWeierstrass> JacobianPoint<E> {
+    pub fn double(&self) -> Self {
+        if self.y == FieldElement::zero() {
+            return Self::neutral_element();
+        }
+        let s = FieldElement::<E::BaseField>::from(4) * &self.x * &self.y.square();
+        let m = FieldElement::<E::BaseField>::from(3) * &self.x.square()
+            + E::a() * &self.z.square().square();
+
+        let two = FieldElement::<E::BaseField>::from(2);
+
+        let x = &m.square() - &two * &s;
+        let y = m * (s - &x) - FieldElement::<E::BaseField>::from(8) * &self.y.square().square();
+        let z = &two * &self.y * &self.z;
+        Self { x, y, z }
+    }
+    /// Creates the same point in affine coordinates. That is,
+    /// returns $[\frac{x}{z^2} : \frac{y}{z^3} : 1]$ where `self` is $[x: y: z]$.
+    /// Panics if `self` is point at infinity.
+    pub fn to_affine(&self) -> Self {
+        let z_sq = self.z.square();
+        Self {
+            x: &self.x / &z_sq,
+            y: &self.y / (&self.z * z_sq),
+            z: FieldElement::one(),
+        }
+    }
+    /// Creates an elliptic curve point giving the Jacobian projective $[x: y: 1]$ coordinates.
+    pub const fn new(
+        x: FieldElement<E::BaseField>,
+        y: FieldElement<E::BaseField>,
+        z: FieldElement<E::BaseField>,
+    ) -> Self {
+        Self { x, y, z }
+    }
```

Comment:
We should make the other new in projectives work like this too, I think it's easier to use. It can be made in a followup